### DISCUSSION TITLE
Capitulo 2 final do Item 2.4 derruba o Thonny

### DISCUSSION THREAD

**Discussion by chicortiz** [clube-de-leitura-python/pense-em-python-2021#13] [Perguntas & Respostas]
Oi pessoal!

Na parte onde temos:

5
x = 5
x + 1

Se o número 5 de cima é removido e tentamos rodar no shell do Thonny o mesmo fica emperrado. Por que será que isso acontece?

O terminal me dá o seguinte:

ortiz@desk:/usr/bin$ ./thonny
INFO: Loading early plugins from ['/usr/bin', '/usr/lib/python36.zip', '/usr/lib/python3.6', '/usr/lib/python3.6/lib-dynload', '/usr/local/lib/python3.6/dist-packages', '/usr/lib/python3/dist-packages', '/home/ortiz/.thonny/plugins/lib/python3.6/site-packages']

Outra coisa curiosa é que derruba o som do meu sistema também... xD

**Comment by villares** [comment]
Nossa que estranho! Você diz, começar pelo x = 5 ?
![image](https://user-images.githubusercontent.com/3694604/129418257-319b7a87-3a9e-45b3-bf23-7eb6cc7b68c4.png)

**Reply by villares** [reply]
eu diria que é algum tipo de bug, o comportamento "normal" não precisaria desse primeiro 5

**Reply by chicortiz** [reply]
> Nossa que estranho! Você diz, começar pelo x = 5 ?
> ![image](https://user-images.githubusercontent.com/3694604/129418257-319b7a87-3a9e-45b3-bf23-7eb6cc7b68c4.png)

Isso mesmo Alê! Remover o primeiro 5 dá xabú para mim. Consegui reproduzir o bug. Deve ser alguma coisa específica do meu sistema que é o Ubuntu 18.04.5 LTS ;)

**Reply by omyasuda** [reply]
Como você instalou o Thonny??

**Reply by chicortiz** [reply]
> Como você instalou o Thonny??

Oi Omyasuda! Creio que foi com  pip install thonny, mas já não me lembro com certeza :I

**Reply by omyasuda** [reply]
Tá me parecendo que esta sua instalação do thonny está quebrada....

que tal tentar instalar diretamente o thonny com

~ $ sudo apt install thonny

??

Acabo de fazer isso aqui no meu Debian, instalou bem, e não apresenta este problema que você reporta!

**Reply by chicortiz** [reply]
Legal Omyasuda! Você recomenda algum processo de desinstalação antes?

**Reply by omyasuda** [reply]
No caso, se o Thonny foi instalado com o pip,

_**pip uninstall thonny**_ 

deve resolver.

**Reply by chicortiz** [reply]
Obrigado Omyasuda mas eu acho que me enganei, não instalei com o pip não deve ter sido sudo apt install mesmo veja:

ortiz@desk:~$ pip uninstall thonny

Command 'pip' not found, but can be installed with:

sudo apt install python-pip

ortiz@desk:~$

**Reply by omyasuda** [reply]
chicortiz, suspeito de alguma incompatibilidade de versão entre o thonny que você tem instalado e o ambiente python.... 

vamos ver como o thonny está insttalado.
**_sudo apt search thonny_**

isso deve mostrar como thonny aparece no seu Ubuntu

**Reply by chicortiz** [reply]
Aí vai a resposta Omyasuda:

ortiz@desk:~$ sudo apt search thonny
[sudo] password for ortiz: 
Sorting... Done
Full Text Search... Done
thonny/bionic,bionic,now 2.1.16-3 all [installed]
  Python IDE for beginners

ortiz@desk:~$

**Reply by omyasuda** [reply]
chicoortiz, parece que temos mesmo algum problema de versão nesta sua instalação usando Ubuntu....

$ sudo apt search thonny
[sudo] senha para omy: 
Sorting... Pronto
Full Text Search... Pronto
thonny/stable,stable,now 3.2.7-1 all [installed]
  Python IDE for beginners

Veja que a sua versão de thonny instalada é bem anterior a que eu instalei aqui no meu Debian....

Alguma chance de atualizarmos o seu Ubuntu? Acho que já estamos na versão 21.04 do Ubuntu.

Sem atualizar o Ubuntu, teremos que ir procurando os pontos de incompatilidade e ir voltando cada um dos programas thonny, pyhton, etc.... o que não me parece muito produtivo.....

**Reply by chicortiz** [reply]
> chicoortiz, parece que temos mesmo algum problema de versão nesta sua instalação usando Ubuntu....
> 
> $ sudo apt search thonny
> [sudo] senha para omy:
> Sorting... Pronto
> Full Text Search... Pronto
> thonny/stable,stable,now 3.2.7-1 all [installed]
> Python IDE for beginners
> 
> Veja que a sua versão de thonny instalada é bem anterior a que eu instalei aqui no meu Debian....
> 
> Alguma chance de atualizarmos o seu Ubuntu? Acho que já estamos na versão 21.04 do Ubuntu.
> 
> Sem atualizar o Ubuntu, teremos que ir procurando os pontos de incompatilidade e ir voltando cada um dos programas thonny, pyhton, etc.... o que não me parece muito produtivo.....

Farei o possível para atualizar meu Ubuntu Omyasuda! Muitíssimo obrigado por seu tempo!

**Comment by chicortiz** [comment]
Aaaah! Eu acho que foi porque eu não limpei completamente o cursor...

**Reply by villares** [reply]
uia! mas é bem terrível isso, pois um espaço ou caratere invisível assim atrapalha muito!
que bom que deu certo no final.
